webapp/components/survey/NodeDefDetails/store/actions/useCancelEdits.js

```diff
@@ -7,30 +7,44 @@ import { NodeDefsActions } from '@webapp/store/survey'
 
 import { State } from '../state'
 
-export const useCancelEdits = () => {
+export const useCancelEdits = ({ setState }) => {
   const history = useHistory()
   const dispatch = useDispatch()
 
   const cancelEdits = ({ state }) => async () => {
     const nodeDef = State.getNodeDef(state)
     const nodeDefOriginal = State.getNodeDefOriginal(state)
 
+    await setState(State.assocNodeDef(nodeDefOriginal))
+
     await dispatch(NodeDefsActions.cancelEdit({ nodeDef, nodeDefOriginal }))
 
     history.goBack()
   }
 
-  return useCallback(async ({ state }) => {
-    const dirty = State.isDirty(state)
-    if (dirty) {
-      dispatch(
-        DialogConfirmActions.showDialogConfirm({
-          key: 'common.cancelConfirm',
-          onOk: cancelEdits({ state }),
-        })
-      )
-    } else {
-      await cancelEdits({ state })()
-    }
-  }, [])
+  return useCallback(
+    async ({ state }) =>
+      new Promise((resolve, reject) => {
+        const dirty = State.isDirty(state)
+        if (dirty) {
+          dispatch(
+            DialogConfirmActions.showDialogConfirm({
+              key: 'common.cancelConfirm',
+              onOk: () => {
+                cancelEdits({ state })()
+                resolve(true)
+              },
+              onCancel: () => {
+                resolve(false)
+              },
+            })
+          )
+        } else {
+          cancelEdits({ state })()
+            .then(() => resolve(true))
+            .catch((e) => reject(e))
```

Comment:
not possible to use async/await ?

Reply:
since the Promise can resolve inside the onOk or onCancel callbacks passed to showDialogConfirm, the only way to resolve it there was by using its methods `then` and `catch`.

Reply:
can't you add a try/catch instead ?

Reply:
I've tried... but there is a problem: `Promise executor functions should not be async`
So, in the end this solution keeps the code a bit more readable.